### PR TITLE
fix : Add check available reserve and adjust default gas

### PR DESCRIPTION
**Pull request by sunspirit99**

### CODE REVIEW

File: pkg/liquidity-source/fluid/vault-t1/pool_simulator.go [hunk 1]
```diff
@@ -116,3 +127,13 @@ func (s *PoolSimulator) CanSwapTo(address string) []string {
 
 	return result
 }
+
+// Helper function to get reserve for a specific token
+func (s *PoolSimulator) getReserveForToken(token string) (*big.Int, error) {
+	for i, t := range s.GetTokens() {
```
Comment (NgoKimPhu) [inline pkg/liquidity-source/fluid/vault-t1/pool_simulator.go:133]: Use GetTokenIndex

File: pkg/liquidity-source/fluid/vault-t1/pool_simulator.go [hunk 2]
```diff
@@ -66,6 +68,15 @@ func (s *PoolSimulator) CalcAmountOut(param poolpkg.CalcAmountOutParams) (*poolp
 
 	tokenAmountOut = new(big.Int).Div(tokenAmountOut, divisor1e27)
 
+	reserveTokenOut, err := s.getReserveForToken(param.TokenOut)
+	if err != nil {
+		return nil, err
+	}
+
+	if tokenAmountOut.Cmp(reserveTokenOut) >= 0 {
```
Comment (NgoKimPhu) [inline pkg/liquidity-source/fluid/vault-t1/pool_simulator.go:76]: Is it not okay to be equal?

File: pkg/liquidity-source/fluid/vault-t1/pool_simulator.go (hunk 1 above)
Comment (sunspirit99) [reply pkg/liquidity-source/fluid/vault-t1/pool_simulator.go]: 8cfe6ed351c9baf235c5c3b46ea44b108d638ed6

File: pkg/liquidity-source/fluid/vault-t1/pool_simulator.go (hunk 2 above)
Comment (sunspirit99) [reply pkg/liquidity-source/fluid/vault-t1/pool_simulator.go:76]: yes, it can be :)) f635c16378ee4e3f71b55d6566ef96f3b82f7e6c

**Review by NgoKimPhu — APPROVED** [review]

**Pull request merged**